### ISSUE TITLE
Template Library - Export with Assigned Template Name - not generic name

### ISSUE THREAD

**Issue by mark2090** [pojome/elementor#5644]
The export/import feature really makes Elementor worth it, to be able to export and import templates between websites. 

Yet there is a problem with the current setup of this feature. 

This refers to templates we create ourselves, this does not refer to pre-made templates. 

For a few templates, like 2-3 at most, it is not a large issue, but the issue grows when there are a lot more templates stored in the library. 

What happens is when a page layout is saved as a template, we name it what we want. The template is saved in the library with that name we gave it. The problem occurs when we export our own templates. The plugin renames these tempates with a generic name as "elementor-(date-more numbers).json".

If we have 4+ templates (in my case 10), then importing those templates and trying to use them becomes a real pain. The pain is when we cannot identity which template is which because the names we assigned to them when we saved them as templates were removed/replaced in the export process. 

Currently, for all imported templates, we have to preview the template to see which is which. This is a lot more work and time wasted trying to find something that should be easy to do. 

The export process is also not a simple click and save. Now we have to rename the export file, for each one we export, which adds more time to exporting. Just so we can know which template is which, the current process is like this, 

- save page as template - named as we choose
- export that same template, renaming it again
- continue to the next page to save as a template
- export that template and rename it again, etc

Now I hope you can see the extra work we end up doing just to find which imported template it was we were seeking. 

**Request**
Change the export process. Please don't remove the names we assigned a template, instead you can append or prepend a date to it. Even append the template name with bracketed "(elementor)" if you want. Just keep the names we assigned the template, so when a template is saved and exported, it retains the same name we gave it in the first place. 

(Note: keep in mind some people make long file names, so truncating a template name, during export, at the spot where most of the name is retained, allowing for easy identification later).

This will then remove the extra work and allow us to identify the template we want to reuse much easier. 

Thanks

**Comment by bainternet** [comment]
@mark2090 

Not sure I understand what the issue or request is, as far as I know, Elementor exports the template name and also imports it, so unless I'm missing something? ...

**Comment by mark2090** [comment]
Have you tried to export a template from the library, a template created by you?

For example, I created a template from a page I made with Elementor and named that template "Contact Form Web Design", which is also the same name as the page itself. 

I exported that template to save on my Desktop and Elementor did not export that template using the same name, Elementor instead renamed the file to "elementor-4925-2018-09-02.json". 

Now imagine 10 or more templates, exported, all renamed like that. At a later time, how is a person going to recognize or remember which template is which? The only means to discern which is which is "preview" each one, which can mean all 10 before finding the template to import elsewhere. 

So my suggestion proposes the following, Elementor exports the template, naming it something similar as follows, "2018-09-02-contact-form-web(elementor).json". 

Essentially, Elementor should export the template keeping a fair porition of the name we gave the template so it can be recognized later, as what page the template refers to.

Does that explain it?

**Comment by mark2090** [comment]
Yes, it exports only 1 type of file, which is a json.

**Comment by Flipback8** [comment]
I agree. This is a much-needed feature. No more Generic names for the exports of the .json files when we have already assigned them our own name. Please fix asap.

**Comment by gerardreches** [comment]
@bainternet Related to #9283
